scripts/cluster/agent.py

```diff
@@ -280,12 +283,14 @@ def join_node():
         port = request.form['port']
         callback_token = request.form['callback']
 
+    # Remove expired tokens
+    remove_expired_token_from_file(cluster_tokens_file)
+
     if not is_valid(token):
         error_msg={"error": "Invalid token"}
         return Response(json.dumps(error_msg), mimetype='application/json', status=500)
 
     add_token_to_certs_request(token)
-    remove_token_from_file(token, cluster_tokens_file)
```

Comment:
Lets keep the old type of tokens for backwards compatibility. The ttl based tokens may be considered unsafe in some cases.